python/ray/data/tests/test_execution_optimizer.py

```diff
@@ -654,6 +691,360 @@ def test_zip_e2e(ray_start_regular_shared, enable_optimizer, num_blocks1, num_bl
     _check_usage_record(["ReadRange", "Zip"])
 
 
+def test_from_dask_operator(ray_start_regular_shared, enable_optimizer):
+    import dask.dataframe as dd
+
+    df = pd.DataFrame({"one": list(range(100)), "two": list(range(100))})
+    ddf = dd.from_pandas(df, npartitions=10)
+
+    planner = Planner()
+    from_dask_op = FromDask(ddf)
+    plan = LogicalPlan(from_dask_op)
+    physical_op = planner.plan(plan).dag
+
+    assert from_dask_op.name == "FromDask"
+    assert isinstance(physical_op, InputDataBuffer)
+    assert len(physical_op.input_dependencies) == 0
+
+
+def test_from_dask_e2e(ray_start_regular_shared, enable_optimizer):
+    import dask.dataframe as dd
+
+    df = pd.DataFrame({"one": list(range(100)), "two": list(range(100))})
+    ddf = dd.from_pandas(df, npartitions=10)
+    ds = ray.data.from_dask(ddf)
+    dfds = ds.to_pandas()
+    assert df.equals(dfds)
+    # Underlying implementation uses `FromPandasRefs` operator
+    assert ds._plan._logical_plan.dag.name == "FromPandasRefs"
```

Comment:
let's also check telemetry is working by calling `_check_usage_record(,,,)` similar to other test cases here.